examples/expander_example.py

```diff
@@ -0,0 +1,56 @@
+import gi
+
+gi.require_version("Gtk", "3.0")
+from gi.repository import Gtk
+
+
+class ExpanderExample(Gtk.Window):
+    def __init__(self):
+        Gtk.Window.__init__(self, title="Expander Demo")
+
+        self.set_size_request(350, 100)
+
+        vbox = Gtk.Box(orientation=Gtk.Orientation.VERTICAL, spacing=6)
+        self.add(vbox)
+
+        text_expander = Gtk.Expander(label="This expander displays additional information")
+        text_expander.set_expanded(True)
+        text_expander.connect('activate', self.on_expand)
+        vbox.add(text_expander)
+
+        msg = """
+This message is quite long, complicated even:
+    - It has a list with a sublist:
+        - of 3 elements;
+        - taking several lines;
+        - with indentation.
+"""
+        details = Gtk.Label(label=msg)
+        text_expander.add(details)
+
+        widget_expander = Gtk.Expander(label="Expand for more controls")
+        widget_expander.connect('activate', self.on_expand)
+        vbox.add(widget_expander)
+
+        expander_hbox = Gtk.HBox()
+        widget_expander.add(expander_hbox)
+
+        expander_hbox.add(Gtk.Label(label="Text message"))
+        expander_hbox.add(Gtk.Button(label="Click me"))
+
+        self.expanders = [text_expander, widget_expander]
+
+        self.show_all()
+
+    def on_expand(self, expander):
+        text_expander, widget_expander = self.expanders
+        if expander == text_expander:
+            widget_expander.set_expanded(False)
+        else:
+            text_expander.set_expanded(False)
```

Comment:
I find it a little bit confusing to hide the widgets of the other expander when the expander is clicked. In particular, because usually use don't need to connect to this signal at all, because hiding/showing done automatically.

The C documentation only mentions it in case you want to create/destroy widgets on-the-fly when expanding/hiding, see https://developer.gnome.org/gtk3/stable/GtkExpander.html

Reply:
Makes sense, thanks!